src/csvcubed/writers/qbwriter.py

```diff
@@ -540,19 +540,19 @@ def _get_primary_key_columns(self) -> List[str]:
         return primary_key_columns
 
     def _determine_whether_column_is_required(self, column: QbColumn) -> bool:
-        return (
-            isinstance(
-                column.structural_definition,
-                (QbDimension, QbMultiUnits, QbMultiMeasureDimension),
-            )
-            or (
-                isinstance(column.structural_definition, QbAttribute)
-                and column.structural_definition.get_is_required()
-            )
-            or (
-                isinstance(column.structural_definition, QbObservationValue)
-                and len(get_observation_status_columns(self.cube)) == 0
-                # We cannot mark an observation value column as `required` if there are `obsStatus` columns defined
-                # since we permit missing observation values where an `obsStatus` explains the reason.
-            )
-        )
+        if isinstance(column.structural_definition,(QbDimension, QbMultiUnits, QbMultiMeasureDimension)):
+            return True
+        elif (isinstance(column.structural_definition, QbAttribute) and column.structural_definition.get_is_required()):
+            if self.cube.is_pivoted_shape:
+                #If the cube is in pivoted shape, the attribute columns cannot be set to required.
+                _logger.warning(
+                    "Setting the column '%s' to not required as the cube is in pivoted shape.", column.csv_column_title
```

Comment:
Can you alter this message to say something like `Attribute column '%s' was marked as required, but the cube has multiple observed values columns. Attributes in multi-measure pivoted cubes cannot currently be marked as required.`